IN-CORE/plotting-service
# Make 3d fragility plot and format the output #1

To test the utility function itself (without spining up the whole app), paste the below code in `util.py`

```
if __name__ == "__main__":
    import json
    from pyincore import FragilityCurveSet

    with open("../test/3d_fragility.json", "r") as f:
        fragility_set = json.load(f)
        xyz_set = get_refactored_xyz_fragility_set(FragilityCurveSet(fragility_set))

    with open("../test/new-fragility-set.json", "r") as f:
        fragility_set = json.load(f)
        xyz_set = get_refactored_xy_fragility_set(FragilityCurveSet(fragility_set))
```

To test the app
- run `python app.py`
- POST to `http://localhost:5000/plotting/api/samples?refresh=true&sample_interval=0.8` with body of the 3d fragility json (you can find it under `/test` folder)
- play with the sample interval, start, end to see differences
